pandas/core/dtypes/cast.py

```diff
@@ -21,15 +22,28 @@
 from .dtypes import (
     DatetimeTZDtype, ExtensionDtype, PandasExtensionDtype, PeriodDtype)
 from .generic import (
-    ABCDatetimeArray, ABCDatetimeIndex, ABCPeriodArray, ABCPeriodIndex,
-    ABCSeries)
+    ABCDatetimeArray, ABCDatetimeIndex, ABCIndexClass, ABCPeriodArray,
+    ABCPeriodIndex, ABCSeries)
 from .inference import is_list_like
 from .missing import isna, notna
 
 _int8_max = np.iinfo(np.int8).max
 _int16_max = np.iinfo(np.int16).max
 _int32_max = np.iinfo(np.int32).max
 _int64_max = np.iinfo(np.int64).max
+_int64_min = np.iinfo(np.int64).min
+_uint64_max = np.iinfo(np.uint64).max
+_float32_max = np.finfo(np.float32).max
+
+
+def _is_iNaT(x):
```

Comment:
is this trying to catch the int64 inat or one/both of numpy's nats?

Reply:
This one is only for the int64. I'm not testing the np-NaTs in #23982 yet, but that could be expanded.

Reply:
Can you add a docstring to this effect for when I inevitably forget this answer

Reply:
done